hal/sparcv8leon3/cpu.c

```diff
@@ -15,9 +15,56 @@
 
 #include "cpu.h"
 
+#define WATCHDOG_CTRL  0x78
+#define DIS_LVDS_ADDR  0x8000d030
+#define CGU_UNLOCK     0x80006000
+#define CGU_EN         0x80006004
+#define CGU_RESET      0x80006008
+#define BO_VMON        0x8010c018
+#define BOOTSTRAP_ADDR 0x80008000
+#define BOOTSTRAP_SPIM 0x400BC003
+#define TESTCFG_ADDR   0x8000E000
+#define TESTCFG_DISR   0x3
+#define SPIM_CTRL_ADDR 0xFFF00104
+#define SPIM_EDAC_ADDR 0xFFF00114
+
+
 void hal_cpuReboot(void)
 {
-	/* TODO */
-	/* *(u32 *)(INT_CTRL_BASE + 128) = 0x00000001; */
-	/* __builtin_unreachable(); */
+	/* Reset to the built-in bootloader */
+
```

Comment:
Perhaps disable interrupts just in case, so reschedule won't ruin something here?